### DISCUSSION TITLE
Lint and autoformat the whole repo with rubocop

### DISCUSSION THREAD

**Discussion by coaxial** [benjaminoakes/maid#220] [Ideas]
I'm thinking of setting up rubocop to lint the repo so that the style is consistent (and autoformatted, either with a git hook or Guard).

This would also imply running rubocop in autofix mode to format all the code properly in a standalone pull request. This would avoid feature PR that include rubocop reformatting the file and showing a diff with many more changes than the actual changes from the feature.

So we'd have one PR and commit like `chore: autoformat code` or something like that, which would have hundreds of lines changed as part of the reformatting. And then run Guard in the background to autoformat files as they are modified, which also means that you can type quick changes this way: `foo=SomeMethod.lala(spaces,and,indent, wont,matter)` which will be prettified when the file is saved.

It would also ensure consistent style across contributors and by using Guard or a precommit hook, there would be no need for bikeshedding or nitpicking across style (beyond the initial rubucop rules setup ;))

I've done it in previous projects and quite liked not having to worry about indents, trailing commas, etc anymore.

@benjaminoakes: What do you think?

**Comment by benjaminoakes** [comment]
I'm definitely in favor of something like that.  I have also had projects that have a GitHub action that enforces style. Another action would actually make a pull request for your pull request with any style changes. We've really only need the former though.

As far as which style guide to follow, a lot of of the code in this repository is much different than code that I would write today.  I'm not very tied to the style in this repository.

That said, I'm working on an open source project at work (hoping to release it very soon!).  It uses standardrb. It would be nice to be consistent and my hope is that it would be less bike shedding.  I'm taking the lead of the doximity/simplekiq developer though.  I don't actually have a ton of experience with standardrb. It seems pretty good so far though.

**Comment by benjaminoakes** [comment]
Oops, wrong button 🙂. Didn't mean to close

**Comment by benjaminoakes** [comment]
I should add that consistent style is important, and I defer to your opinion as the most active developer as far as which style to follow.

**Comment by coaxial** [comment]
Alright, I have a boilerplate rubocop rules file for Rails projects that I will need to adapt.

**Reply by benjaminoakes** [reply]
Sounds good. However, please read my full comment which was not sent fully over email. (Sorry about that.)

**Comment by coaxial** [comment]
Done in #224
